Header: Switch to submenu blocks for navigation items with submenus

ryelle opened the pull request:
Fixes #207 - Menu items with submenus should use the `navigation-submenu` block, this provides the expected markup for accessibility:

- Submenus don't open on focus.
- Items with a submenu (Support & Get Involved) have buttons following that say "Support submenu"/"Get Involved submenu".
- The button has `aria-expanded="false"` by default, and `true` when open

The remaining item on the list in #207 is this:

> If inside a sub menu, pressing Escape should close the sub menu and move focus to the trigger that opened it.

This menu doesn't seem to do that, but this doesn't seem to be core behavior either. I'd recommend opening an issue in gutenberg for the navigation block if this is required.

Props @alexstine.

**Screenshots**

In both of the videos below, I tab through the menu. In "before" you can see I tab down into the submenus with no other action. After, I need to activate the 🔽  button. There's no visual change, and no change in behavior with a mouse.

https://user-images.githubusercontent.com/541093/174152708-901a580b-468d-4401-93a0-a441ca8d3d74.mp4

https://user-images.githubusercontent.com/541093/174152715-28aa70fb-62ac-4ddb-ad4d-f7dc1aefccce.mp4

| Before | After |
|--------|------|
| ![](https://user-images.githubusercontent.com/541093/174153340-6f4348be-0905-4551-b1a2-effd20c3505e.png) | ![](https://user-images.githubusercontent.com/541093/174153336-123a9fb3-d5f1-4790-839f-8f6fe0e62a87.png) |
| ![](https://user-images.githubusercontent.com/541093/174153343-15b0caf9-9f8c-4e25-9538-c6451f80c46f.png) | ![](https://user-images.githubusercontent.com/541093/174153339-8913c273-2e62-4991-a706-82a16e41ad6d.png) |

ryelle committed:
Header: Switch to submenu blocks for navigation items with submenus

Fixes #207

StevenDufresne commented:
When the menu is truncated, the submenu is not reachable:

https://user-images.githubusercontent.com/1657336/174197467-5b3d5b14-fe4e-49ca-a049-5c1cf9ba1e95.mp4

StevenDufresne commented on mu-plugins/blocks/global-header-footer/header.php:
```diff
@@ -26,6 +27,45 @@
+		);
+	}
+
+	$ouput = sprintf(
```
I think this is a typo: `$output` maybe?

ryelle replied:
🤦🏻

alexstine commented:
Thanks @ryelle ! I opened a PR upstream in Gutenberg which should fix up Escape key functionality.

ryelle committed:
Fix typo

ryelle committed:
Display second level items inline in the overflow menu

ryelle committed:
Use the toggle button pattern for the overflow menu

ryelle commented:
> When the menu is truncated, the submenu is not reachable

Ah, I bet when the menu item is cloned, the toggle looses the event listener. I've updated the style on submenus in the overflow menu so they'll always display to work around that. Also because 3rd-level flyouts are awkward :)

![](https://user-images.githubusercontent.com/541093/174680410-8a530450-9fff-4297-b66a-8c4ea1db71dc.png)

I also updated the `...` link when the menu is truncated to be a button and use the same toggle-on-hover/click pattern. This fixes an issue where screen readers would just announce the link as "dot" (at least, VoiceOver did), now it's "More menu" with the expanded context.

StevenDufresne reviewed (approved):
Great, looks good.
